lib/ansible/modules/cloud/cloudstack/cs_physical_network.py

```diff
@@ -0,0 +1,470 @@
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible. If not, see <http://www.gnu.org/licenses/>.
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: cs_physical_network
+short_description: Manages physical networks on Apache CloudStack based clouds.
+description:
+    - Create, update and remove networks.
+    - Enabled and disabled Network Service Providers
+    - Enables Internal LoadBalancer and VPC/VirtualRouter elements as required
+version_added: "2.8"
+author:
+ - Netservers Ltd. (@netservers)
+ - Patryk Cichy (@PatTheSilent)
+options:
+  physical_network:
+    description:
+      - Name of the physical network.
+    required: true
+    aliases:
+      - name
+  zone:
+    description:
+      - Name of the zone in which the network belongs.
+      - If not set, default zone is used.
+    required: true
+  broadcast_domain_range:
+    description:
+      - broadcast domain range for the physical network[Pod or Zone].
+    required: false
+    choices: [ POD, ZONE ]
+  domain:
+    description:
+      - Domain the network is owned by.
+    required: false
+  isolation_method:
+    description:
+      - Isolation method for the physical network.
+    required: false
+    choices: [ VLAN, GRE, L3 ]
+  network_speed:
+    description:
+      - The speed for the physical network.
+    required: false
+    choices: [1G, 10G]
+  tags:
+    description:
+      - A tag to identify this network.
+      - Physical networks support only one tag.
+      - To remove an existing tag pass an empty string.
+    required: false
+    aliases:
+      - tag
+  vlan:
+    description:
+      - The VLAN/VNI Ranges of the physical network.
+    required: false
+  nsps_enabled:
+    description:
+      - List of Network Service Providers to enable
+    required: false
+  nsps_disabled:
+    description:
+      - List of Network Service Providers to disable
+    required: false
+  state:
+    description:
+      - State of the physical network.
+    required: false
+    default: present
+    choices: [ present, absent, disabled, enabled ]
+  poll_async:
+    description:
+      - Poll async jobs until job has finished.
+    required: false
+    default: true
+extends_documentation_fragment: cloudstack
+'''
+
+EXAMPLES = '''
+# Ensure a network is present
+- local_action:
```

Comment:
For consistency, please use the `delegate_to: localhost` syntax.

Reply:
Done.